pkg/clustermanager/cluster_creator_test.go

```diff
@@ -0,0 +1,69 @@
+package clustermanager_test
+
+import (
+	"os"
+	"testing"
+	"time"
+
+	"github.com/golang/mock/gomock"
+	. "github.com/onsi/gomega"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/pointer"
+
+	corev1 "k8s.io/api/core/v1"
+
+	"github.com/aws/eks-anywhere/internal/test"
+	"github.com/aws/eks-anywhere/pkg/clients/kubernetes"
+	"github.com/aws/eks-anywhere/pkg/clustermanager"
+	"github.com/aws/eks-anywhere/pkg/constants"
+	"github.com/aws/eks-anywhere/pkg/controller/clientutil"
+	mockswriter "github.com/aws/eks-anywhere/pkg/filewriter/mocks"
+	"github.com/aws/eks-anywhere/pkg/kubeconfig"
+)
+
+type clusterCreatorTest struct {
+	*WithT
+	*applierTest
```

Comment:
why don't you decouple the creator from the applier?
That way you don't need use this applier test helpers. Just mock the Applier.Run call.